include/context.h

```diff
@@ -22,6 +22,12 @@ typedef enum {
 	R_CONTEXT_CONFIG_MODE_REQUIRED, /* require config file */
 } RContextConfigMode;
 
+typedef struct {
+	gchar *overwrite_section;
+	gchar *overwrite_name;
+	gchar *overwrite_value;
+} ConfigFileOverwrite;
+
```

Comment:
I'd use [override instead of overwrite](https://www.grammar.com/override_vs_overwrite). Also there's no need to repeat prefix all structure members with the same string.

```suggestion
typedef struct {
	gchar *section;
	gchar *name;
	gchar *value;
} ConfigFileOverride;
```

ee